src/OrbitQt/orbithistogram.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "orbithistogram.h"
+
+#include <absl/strings/str_cat.h>
+#include <absl/strings/str_format.h>
+#include <qevent.h>
+#include <qnamespace.h>
+#include <qpainter.h>
+#include <qpoint.h>
+#include <qwidget.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <iterator>
+#include <string>
+#include <utility>
+
+#include "DisplayFormats/DisplayFormats.h"
+#include "Statistics/Histogram.h"
+
+constexpr double kRelativeMargin = 0.1;
+
+constexpr uint32_t kTicksNum = 3;
+constexpr int kTickLength = 5;
+
+[[nodiscard]] static int RoundToClosestInt(double x) { return static_cast<int>(std::round(x)); }
+
+[[nodiscard]] static int Width(const QPainter& painter) { return painter.device()->width(); }
+
+[[nodiscard]] static int WidthMargin(const QPainter& painter) {
+  return RoundToClosestInt(Width(painter) * kRelativeMargin);
+}
+
+[[nodiscard]] static int Height(const QPainter& painter) { return painter.device()->height(); }
+
+[[nodiscard]] static int HeightMargin(const QPainter& painter) {
+  return RoundToClosestInt(Height(painter) * kRelativeMargin);
+}
+
+// if `length > 0`, the line will be plot to the right from `start` and to the left otherwise
+static void DrawHorizontalLine(QPainter& painter, const QPoint& start, int length) {
+  painter.drawLine(start, {start.x() + length, start.y()});
+}
+
+// if `length > 0`, the line will be plot downwards from `start` and upwards otherwise
+static void DrawVerticalLine(QPainter& painter, const QPoint& start, int length) {
+  painter.drawLine(start, {start.x(), start.y() + length});
+}
+
+static void DrawHorizontalAxis(QPainter& painter, const QPoint& zero,
```

Comment:
Nit: Maybe origin or offset instead of zero?

Reply:
Maybe `axes_itersection`?

Reply:
Yes would also work for me.